plutus-core/plutus-ir/src/PlutusIR/Transform/RewriteRules.hs

```diff
@@ -0,0 +1,59 @@
+{-# LANGUAGE FlexibleInstances #-}
+{-# LANGUAGE GADTs             #-}
+{-# LANGUAGE RankNTypes        #-}
+{-# LANGUAGE TypeOperators     #-}
+module PlutusIR.Transform.RewriteRules
+    ( userRewrite
+    , RewriteRules (..)
+    , defaultUniRewriteRules
+    ) where
+
+import PlutusCore.Default
+import PlutusCore.Name
+import PlutusCore.Quote
+import PlutusIR as PIR
+import PlutusIR.Analysis.VarInfo
+import PlutusIR.Transform.RewriteRules.CommuteFnWithConst
+import PlutusPrelude
+
+import Control.Lens
+
+
+-- | Rewrite a `Term` using the given `RewriteRules` (similar to functions of Term -> Term)
+-- Normally the rewrite rules are configured at entrypoint time of the compiler.
+userRewrite :: ( Semigroup a, t ~ Term tyname name uni fun a
+              , HasUnique name TermUnique
+              , HasUnique tyname TypeUnique
```

Comment:
You can say `HasUniques t` instead of `HasUnique name TermUnique, HasUnique tyname TypeUnique`.